x-pack/plugins/cases/public/components/recent_cases/recent_cases.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiText } from '@elastic/eui';
+import React, { useEffect, useMemo, useRef } from 'react';
+import { isEqual } from 'lodash/fp';
+import styled from 'styled-components';
+
+import { IconWithCount } from './icon_with_count';
+import * as i18n from './translations';
+import { useGetCases } from '../../containers/use_get_cases';
+import { CaseDetailsHrefSchema, CaseDetailsLink } from '../links';
+import { LoadingPlaceholders } from './loading_placeholders';
+import { NoCases } from './no_cases';
+import { isSubCase } from '../all_cases/helpers';
+import { MarkdownRenderer } from '../markdown_editor';
+import { FilterOptions } from '../../containers/types';
+
+const MarkdownContainer = styled.div`
+  max-height: 150px;
+  overflow-y: auto;
+  width: 300px;
+`;
+
+export interface RecentCasesProps {
+  createCaseHref: string;
+  filterOptions: Partial<FilterOptions>;
+  getCaseDetailsHref: (caseDetails: CaseDetailsHrefSchema) => string;
+  onCaseDetailsNavClick: (caseDetails: CaseDetailsHrefSchema) => void;
+  perPage: number;
+}
+const usePrevious = (value: Partial<FilterOptions>) => {
+  const ref = useRef();
+  useEffect(() => {
+    (ref.current as unknown) = value;
+  });
```

Comment:
This will run on every render, right?

Reply:
i think so, copy/pasted over from @andrew-goldstein